src/instrumentationESM.ts

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License.
+
+// Import Azure Functions types
+import type * as AzFunc from '@azure/functions';
+// Import OpenTelemetry APIs for context propagation
+import { context as otelContext, propagation } from '@opentelemetry/api';
+// Import OpenTelemetry severity levels for logging
+import { SeverityNumber } from '@opentelemetry/api-logs';
+// Import OpenTelemetry instrumentation base classes
+import {
+    InstrumentationBase,
+    InstrumentationConfig,
+    InstrumentationNodeModuleDefinition,
+} from '@opentelemetry/instrumentation';
+// Import instrumentation version constant
+import { version } from './constants';
+
+/**
+ * Azure Functions Instrumentation class specifically designed for ESM scenarios.
+ * Extends OpenTelemetry's InstrumentationBase to integrate with Azure Functions.
+ */
+export class AzureFunctionsInstrumentationESM extends InstrumentationBase {
+    // Array to store disposables for cleanup during unpatching
+    private azFuncDisposable: AzFunc.Disposable[] = [];
+
+    /**
+     * Constructor initializes the instrumentation with a given configuration.
+     * @param config Optional instrumentation configuration.
+     */
+    constructor(config: InstrumentationConfig = {}) {
+        super('@azure/functions-opentelemetry-instrumentation', version, config);
+    }
+
+    /**
+     * Initializes the instrumentation by defining how to patch and unpatch the Azure Functions module.
+     * This method is called automatically by OpenTelemetry instrumentation lifecycle.
+     */
+    protected init() {
+        const instrumentationNodeModuleDefinition = new InstrumentationNodeModuleDefinition(
+            '@azure/functions', // Target module to instrument
+            ['^4.5.0'], // Supported versions
+            (moduleExports: typeof AzFunc) => this._patch(moduleExports), // Patch method
+            (moduleExports: typeof AzFunc) => this._unPatch(moduleExports) // Unpatch method
+        );
+
+        return instrumentationNodeModuleDefinition;
+    }
+
+    /**
+     * Internal method to patch Azure Functions SDK.
+     * Sets up logging and trace context propagation hooks.
+     * @param azFunc Azure Functions module exports.
+     * @returns Patched Azure Functions module exports.
+     */
+    private _patch(azFunc: typeof AzFunc): typeof AzFunc {
+        // Inform Azure Functions Host that logs will be sent directly from Node.js
+        azFunc.app.setup({
+            capabilities: {
+                WorkerOpenTelemetryEnabled: true,
+            },
+        });
+
+        // Hook into Azure Functions logging to emit logs via OpenTelemetry
+        const logDisposable = azFunc.app.hook.log((context) => {
+            this.logger.emit({
```

Comment:
Where is the `logger` defined? 
I don't see logger defined in [InstrumentationBase](https://github.com/open-telemetry/opentelemetry-js/blob/55f8c20b19777602f993328fe07f523cf465dea3/experimental/packages/opentelemetry-instrumentation/src/platform/node/instrumentation.ts#L42)

Reply:
Logger is defined in the abstract class 'InstrumentationAbstract'.

https://github.com/open-telemetry/opentelemetry-js/blob/f2cfd1327a5b131ea795301b10877291aac4e6f5/experimental/packages/opentelemetry-instrumentation/src/instrumentation.ts#L96